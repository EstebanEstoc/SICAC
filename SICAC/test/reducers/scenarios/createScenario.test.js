import createScenarioReducer from '../../../reducers/scenarios/createScenarioSlice'
import {
  addAction,
  removeAction,
  addCondition,
  removeCondition,
  clear,
  addName
} from '../../../reducers/scenarios/createScenarioSlice'

describe('createScenario Reducer', () => {
  describe('INITIAL_STATE', () => {
    test('is correct', () => {
      const action = { type: 'dummy_action' }
      const initialState = { name: '', conditions: [], actions: [] }
      expect(createScenarioReducer(undefined, action)).toEqual(initialState)
    })
  })

  describe('addAction', () => {
    test('must push an action in the then field', () => {
      const actionScenario = {
        id: 33,
        options: {
          lat: -123,
          lng: 13
        }
      }
      const action = { type: addAction.type, payload: actionScenario }

      expect(createScenarioReducer(undefined, action)).toMatchSnapshot()
    })
  })

  describe('addCondition', () => {
    test('must push an condition in the if field', () => {
      const conditionScenario = {
        id: 35,
        options: {
          lat: -123,
          lng: 13
        }
      }
      const action = { type: addCondition.type, payload: conditionScenario }

      expect(createScenarioReducer(undefined, action)).toMatchSnapshot()
    })
  })

  describe('removeCondition', () => {
    test('must delete a condition in the if field', () => {
      const conditionScenario = {
        id: 35,
        options: {
          lat: -123,
          lng: 13
        }
      }
      const addAction = { type: addCondition.type, payload: conditionScenario }
      const removeAction = { type: removeCondition.type, payload: 0 }

      expect(
        createScenarioReducer(
          createScenarioReducer(undefined, addAction),
          removeAction
        )
      ).toMatchSnapshot()
    })
  })

  describe('clear', () => {
    test('must reset to initialState', () => {
      const action = { type: clear.type }

      expect(createScenarioReducer(undefined, action)).toMatchSnapshot()
    })
  })

  describe('addName', () => {
    test('must add name to the name field', () => {
      const name = "toto's scenario"
      const action = { type: addName.type, payload: name }

      expect(createScenarioReducer(undefined, action)).toMatchSnapshot()
    })
  })
})
