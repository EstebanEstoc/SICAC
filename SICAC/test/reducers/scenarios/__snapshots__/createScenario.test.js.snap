// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createScenario Reducer addAction must push an action in the then field 1`] = `
Object {
  "actions": Array [
    Object {
      "id": 33,
      "options": Object {
        "lat": -123,
        "lng": 13,
      },
    },
  ],
  "conditions": Array [],
  "name": "",
}
`;

exports[`createScenario Reducer addCondition must push an condition in the if field 1`] = `
Object {
  "actions": Array [],
  "conditions": Array [
    Object {
      "id": 35,
      "options": Object {
        "lat": -123,
        "lng": 13,
      },
    },
  ],
  "name": "",
}
`;

exports[`createScenario Reducer addName must add name to the name field 1`] = `
Object {
  "actions": Array [],
  "conditions": Array [],
  "name": "toto's scenario",
}
`;

exports[`createScenario Reducer clear must reset to initialState 1`] = `
Object {
  "actions": Array [],
  "conditions": Array [],
  "name": "",
}
`;
